server/orders.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+const db = require('APP/db')
+const User = db.model('users')
+const Order = db.model('orders')
+const Cart = db.model('carts')
+const CartDetail = db.model('cartDetail')
+const Product = db.model('products')
+
+const { mustBeLoggedIn, assertAdmin, selfOnly } = require('./auth.filters')
+
+module.exports = require('express').Router()
+
+  // get all orders (Logged in User, ADMIN)
+  .get('/',
+  mustBeLoggedIn,
+  (req, res, next) => {
+    if (req.user.isAdmin) {
+      if (req.query.status) {
+        Order.findAll({
+          where: {
+            status: req.query.status
+          }
+        }).then(statusOrders => {
+          res.json(statusOrders)
+        })
+      } else {
+        Order.findAll({})
+          .then(orders => {
+            res.json(orders)
+          })
+      }
+    } else {
+      Order.findAll({
+        where: {
+          user_id: req.user.id
+        }
+      }).then(orders => {
+        res.json(orders)
+      })
+    }
+  })
+
+  .post('/', (req, res, next) => {
+    Cart.findOne({
+      where: {
+        user_id: req.user.id
+      },
+      include: [{
+        model: Product
+      }]
+    }).then(order => {
+      return res.json(order)
+    }).then(cart => Order.create({
+      status: 'received',
+      user_id: req.user.id
+    })).catch(next)
+  })
+
+  .param('oid', (req, res, next, oid) => {
```

Comment:
When you look for an Order, consider including Product? That way we can see all the products within that order